kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/ResolutionResult.java

```diff
@@ -36,18 +39,84 @@ public class ResolutionResult {
     private final Map<LocalRef<KafkaService>, KafkaService> kafkaServiceRefs;
     private final Set<ClusterResolutionResult> clusterResolutionResults;
 
-    public record ClusterResolutionResult(VirtualKafkaCluster cluster, Set<LocalRef<?>> unresolvedDependencySet) {
+    public Optional<ClusterResolutionResult> clusterResult(VirtualKafkaCluster cluster) {
+        return clusterResolutionResults.stream()
+                .filter(r -> r.cluster == cluster)
+                .findFirst();
+    }
+
+    enum DependencyType {
+        // from the virtual kafka cluster to another entity
+        DIRECT,
+        // from any other entity to another entity
+        TRANSITIVE
+    }
+
+    public record UnresolvedReferences(Set<UnresolvedReference> unresolved,
+                                       Set<KafkaProxyIngress> ingressesWithResolvedRefsFalse,
+                                       Set<KafkaProtocolFilter> filtersWithResolvedRefsFalse,
+                                       Set<KafkaService> kafkaServicesWithResolvedRefsFalse) {
```

Comment:
this implementation was driven more by the VirtualKafkaCluster's current code, where we want to know about each type of problem to generate a different `reasons` on the `ResolvedRefs` condition. Maybe it would simplify things if we adopted the KafkaProxyReconciler style where we emit a message like "resource <slug> not found" or "resource <slug> has ResolvedRefs: False condition".